stdsimd-verify/tests/x86-intel.rs

```diff
@@ -0,0 +1,299 @@
+    ret: Option<&'static Type>,
+    target_feature: &'static str,
+    instrs: &'static [&'static str],
+}
+
+static BOOL: Type = Type::Bool;
+static F32: Type = Type::PrimFloat(32);
+static F32x4: Type = Type::Float(32, 4);
+static F32x8: Type = Type::Float(32, 8);
+static F64: Type = Type::PrimFloat(64);
+static F64x2: Type = Type::Float(64, 2);
+static F64x4: Type = Type::Float(64, 4);
+static I16: Type = Type::PrimSigned(16);
+static I16x16: Type = Type::Signed(16, 16);
+static I16x4: Type = Type::Signed(16, 4);
+static I16x8: Type = Type::Signed(16, 8);
+static I32: Type = Type::PrimSigned(32);
+static I32x2: Type = Type::Signed(32, 2);
+static I32x4: Type = Type::Signed(32, 4);
+static I32x8: Type = Type::Signed(32, 8);
+static I64: Type = Type::PrimSigned(64);
+static I64x2: Type = Type::Signed(64, 2);
+static I64x4: Type = Type::Signed(64, 4);
+static I8: Type = Type::PrimSigned(8);
+static I8x16: Type = Type::Signed(8, 16);
+static I8x32: Type = Type::Signed(8, 32);
+static I8x8: Type = Type::Signed(8, 8);
+static U16: Type = Type::PrimUnsigned(16);
+static U16x16: Type = Type::Unsigned(16, 16);
+static U16x4: Type = Type::Unsigned(16, 4);
+static U16x8: Type = Type::Unsigned(16, 8);
+static U32: Type = Type::PrimUnsigned(32);
+static U32x2: Type = Type::Unsigned(32, 2);
+static U32x4: Type = Type::Unsigned(32, 4);
+static U32x8: Type = Type::Unsigned(32, 8);
+static U64: Type = Type::PrimUnsigned(64);
+static U64x2: Type = Type::Unsigned(64, 2);
+static U64x4: Type = Type::Unsigned(64, 4);
+static U8: Type = Type::PrimUnsigned(8);
+static U8x16: Type = Type::Unsigned(8, 16);
+static U8x32: Type = Type::Unsigned(8, 32);
+static U8x8: Type = Type::Unsigned(8, 8);
+
+#[derive(Debug)]
+enum Type {
+    Float(u8, u8),
+    PrimFloat(u8),
+    PrimSigned(u8),
+    PrimUnsigned(u8),
+    Ptr(&'static Type),
+    Signed(u8, u8),
+    Unsigned(u8, u8),
+    Bool,
+}
+
+x86_functions!(static FUNCTIONS);
+
+#[derive(Deserialize)]
+struct Data {
+    #[serde(rename = "intrinsic", default)]
+    intrinsics: Vec<Intrinsic>,
+}
+
+#[derive(Deserialize)]
+struct Intrinsic {
+    rettype: String,
+    name: String,
+    tech: String,
+    #[serde(rename = "CPUID", default)]
+    cpuid: Vec<String>,
+    #[serde(rename = "parameter", default)]
+    parameters: Vec<Parameter>,
+    #[serde(default)]
+    instruction: Vec<Instruction>,
+}
+
+#[derive(Deserialize)]
+struct Parameter {
+    #[serde(rename = "type")]
+    type_: String,
+}
+
+#[derive(Deserialize)]
+struct Instruction {
+    name: String,
+}
+
+#[test]
+fn verify_all_signatures() {
+    let xml = include_bytes!("../x86-intel.xml");
+    let xml = &xml[..];
+    let data: Data = serde_xml_rs::deserialize(xml).expect("failed to deserialize xml");
+    let mut map = HashMap::new();
+    for intrinsic in data.intrinsics.iter() {
+        // This intrinsic has multiple definitions in the XML, so just ignore it.
+        if intrinsic.name == "_mm_prefetch" {
+            continue
```

Comment:
The `_mm_prefetch` intrinsics does something different if the target has AVX-512, SSE, or not even SSE. I think it is ok to ignore these for now (and AVX-512 overloads some intrinsics to have slightly different behavior).